File: src/main/java/com/travel/marketplace/modules/provider/service/AdminProviderService.java
package com.travel.marketplace.modules.provider.service;

import com.travel.marketplace.modules.provider.dto.AdminProviderCategoryResponse;
import com.travel.marketplace.modules.provider.dto.AdminProviderGrowthResponse;
import com.travel.marketplace.modules.provider.dto.AdminProviderResponse;
import com.travel.marketplace.modules.provider.dto.AdminProviderSearchRequest;
import com.travel.marketplace.modules.provider.dto.AdminProviderStatisticsResponse;
import com.travel.marketplace.modules.provider.dto.ProviderProfileResponse;
import org.springframework.data.domain.Page;

import java.util.List;

public interface AdminProviderService {

    Page<AdminProviderResponse> getProviders(AdminProviderSearchRequest request, int page, int size, String sort);

    AdminProviderStatisticsResponse getStatistics();

    List<AdminProviderCategoryResponse> getCategoryDistribution();

    AdminProviderGrowthResponse getGrowth(String range);

    List<AdminProviderResponse> getTopRated(int limit);

    ProviderProfileResponse approveProvider(Long providerId);

    ProviderProfileResponse rejectProvider(Long providerId, String reason);

    ProviderProfileResponse suspendProvider(Long providerId, String reason);

    ProviderProfileResponse reactivateProvider(Long providerId);
}

File: src/services/adminProviderService.ts
import api from './api';
import { ApiResponse, PageResponse } from '@/types';
import {
  AdminProvider,
  AdminProviderCategoryDistribution,
  AdminProviderGrowth,
  AdminProviderSearchParams,
  AdminProviderStatistics,
} from '@/types/adminProvider';
import { ProviderProfileResponse } from '@/types/provider';

export const adminProviderService = {
  async getProviders(params: AdminProviderSearchParams) {
    const response = await api.get<ApiResponse<PageResponse<AdminProvider>>>('/admin/providers', { params });
    return response.data;
  },

  async getStatistics() {
    const response = await api.get<ApiResponse<AdminProviderStatistics>>('/admin/providers/statistics');
    return response.data;
  },

  async getGrowth(range = '30d') {
    const response = await api.get<ApiResponse<AdminProviderGrowth>>('/admin/providers/growth', { params: { range } });
    return response.data;
  },

  async getCategoryDistribution() {
    const response = await api.get<ApiResponse<AdminProviderCategoryDistribution[]>>('/admin/providers/categories');
    return response.data;
  },

  async getTopRated(limit = 5) {
    const response = await api.get<ApiResponse<AdminProvider[]>>('/admin/providers/top-rated', { params: { limit } });
    return response.data;
  },

  async verifyProvider(id: number) {
    const response = await api.patch<ApiResponse<ProviderProfileResponse>>(`/admin/providers/${id}/verify`);
    return response.data;
  },

  async rejectProvider(id: number, reason: string) {
    const response = await api.patch<ApiResponse<ProviderProfileResponse>>(`/admin/providers/${id}/reject`, undefined, { params: { reason } });
    return response.data;
  },

  async suspendProvider(id: number, reason?: string) {
    const response = await api.patch<ApiResponse<ProviderProfileResponse>>(`/admin/providers/${id}/suspend`, undefined, { params: { reason } });
    return response.data;
  },

  async reactivateProvider(id: number) {
    const response = await api.patch<ApiResponse<ProviderProfileResponse>>(`/admin/providers/${id}/reactivate`);
    return response.data;
  },
};
